### ISSUE TITLE
Support for automatic removal of user if email validation token expires and user is unable to activate it during 24 hours

### ISSUE THREAD

**Issue by samip5** [goauthentik/authentik#1868]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when users try to create an user, and activate it using the link but due to the how it currently works, it needs to be the same session, and if user has already destroyed the session, they're unable to use the token at all.

**Describe the solution you'd like**
I would like an automatic way to remove inacactive users that have tried to use email validation, but it has failed.

**Additional context**
Synapse Matrix, and using Authentik as IdP for it

**Comment by BeryJu** [comment]
I would reword/extend this request into a full Recycle bin, either as custom base model for User and Group, or as extension/modification of ExpiringModel.

This can then also be integrated with the LDAP Source, and objects removed in LDAP will be placed into the recylce bin in authentik. 

It would also work with just an attribute set on the user, like `goauthentik.io/core/recycle: true`, that would limit this to User and Groups and make it harder to keep generic.

**Comment by samip5** [comment]
I'm not exactly sure what you mean by a full recycle bin and how does it relate to my original FR?

**Comment by BeryJu** [comment]
After some more thought, it would indeed contradict with the way this recycle bin would work.

For this issue, I think the easiest issue would be to have `User` inherit `ExpiringModel`, which would allow the email stage to set an expiration of `now + token_validity + x` . The main question after the user expires becomes, does the user get deleted or just set to inactive. Obviously for this usecase, deletion makes more sense, but in most other implementations of expiring users, the users are set to inactive.

**Comment by samip5** [comment]
In an ideal case, I would at least like it to be deleted so that the user can re-try the enrollment which wouldn't work if the same user exists but is inactive.

**Comment by samip5** [comment]
Noooo. Bad Stale bot.

**Comment by samip5** [comment]
@BeryJu Could this be re-opened?
